File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    check, count = {}, {i : 0 for i in range(n)}
    check[words[0]] = 1
    count[0] += 1
    
    for i in range(1, len(words)):
        idx, w = i % n, words[i]
        count[idx] += 1
        if check.get(w) or words[i][0] != words[i - 1][-1]:
            return [idx + 1, count[idx]]
        check[w] = 1
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        Map<String, Integer> map = new HashMap<>();
        map.put(words[0], 1);
        
        for(int i = 1; i < words.length; i++) {
            if (map.containsKey(words[i]) || words[i].charAt(0) != words[i - 1].charAt(words[i - 1].length() - 1)) {
                return new int[] {i % n + 1, (int) i / n + 1};
            }
            map.put(words[i], 1);
        }

        return new int[] {0, 0};
    }
}